python/pyarrow/serialization.py

```diff
@@ -37,11 +37,22 @@
 # python_to_arrow.cc)
 
 def _serialize_numpy_array_list(obj):
-    return obj.tolist(), obj.dtype.str
+    if obj.dtype.str != '|O':
```

Comment:
I'm not sufficiently familiar with numpy dtypes to know if this `|O` is the only relevant special case.

Reply:
I think you could check here for `obj.dtype.hasobject`: https://docs.scipy.org/doc/numpy/reference/generated/numpy.dtype.hasobject.html